src/utils/time-utils.ts

```diff
@@ -0,0 +1,107 @@
+import {
+    CAPTURE_INTERVAL_RE,
+    DURATION_RE,
+    POSTGRES_INTERVAL_RE,
+} from 'validation';
+import { hasLength } from './misc-utils';
+
+interface SegmentedInterval {
+    h: number;
+    m: number;
+    s: number;
+}
+
+const parsePostgresInterval = (interval: string): SegmentedInterval => {
+    const [hours, minutes, seconds] = interval.split(':').map((segment) => {
+        const numericSegment = Number(segment);
+
+        return isFinite(numericSegment) ? numericSegment : -1;
+    });
+
+    return { h: hours, m: minutes, s: seconds };
+};
+
+const parseCaptureInterval = (interval: string): SegmentedInterval => {
+    let hours = -1;
+    let minutes = -1;
+    let seconds = -1;
+
+    interval.split(' ').forEach((segment) => {
+        const numericSegment = Number(segment.substring(0, segment.length - 1));
+        const numericValue = isFinite(numericSegment) ? numericSegment : -1;
+
+        if (segment.includes('h')) {
+            hours = numericValue;
+        } else if (segment.includes('m')) {
+            minutes = numericValue;
+        } else if (segment.includes('s')) {
+            seconds = numericValue;
+        }
+    });
+
+    return { h: hours, m: minutes, s: seconds };
+};
+
+export const getCaptureIntervalSegment = (interval: string, unit: string) => {
+    let truncatedInterval = -1;
+
+    const granularTimeUnit = unit === 'h' || unit === 'm' || unit === 's';
+
+    if (granularTimeUnit && POSTGRES_INTERVAL_RE.test(interval)) {
+        const segments = parsePostgresInterval(interval);
+
+        truncatedInterval = segments[unit];
```

Comment:
I think we can just make this and the line below a single line.

Reply:
I'm not the biggest fan of indexing a function's return argument directly but sure, if that is your preference.